frontend/ui/widget/configdialog.lua

```diff
@@ -90,6 +90,8 @@ function OptionTextItem:onTapSelect()
     self.config:onConfigChoose(self.values, self.name,
                     self.event, self.args,
                     self.events, self.current_item, self.hide_on_apply)
+
+    Notification:notify(self.notification_func and self:notification_func())
```

Comment:
May be (sorry to not have thought about it when I suggested it) `notify_func` rather than `notification_func` to make it clear it's used with notify.
Also, this function call (which will do nothing) is uneeded. It shouldn't cost much in this context, but still :) This is not really much longer:
```diff
- Notification:notify(self.notification_func and self:notification_func())
+ if self.notification_func then Notification:notify(self:notification_func()) end
```
But if you want to keep it as it is, in :notify(), check for `arg` (being not nil) before doing the mask and souce work.

Reply:
Please look on my next commit. With this the "decrease font size" gesture shows the message "XXXXX test". :)

Reply:
Maybe the order should be the other way round, the dispatcher message first.
Maybe we can add another "SOURCE_EVENTS" for all notifications by events, so we can distinguish between dispatcher messages and event messages and bottom_menu messages ....

Reply:
> Please look on my next commit.

It just adds a way to add a notification for handlers that don't do that themselves, right?
But it does not solve the fact that kopt bottom menu that don't work with an event handler can't be used by Dispatcher. If we find (later :) a way to solve this, we might no longer need this notification_func.

Reply:
There are the advantages given by `Notification:notify()`.
I was talking about the `notification_func()` you added some hours ago, which just solves the Use case 0 above.

Reply:
Your are right it's not implementen by now, but it can easily be done.
 
Dispatcher shows a `notify` if there is a `notify_func` with SOURCE_DISPATCHER and then emits the events with SOURCE_EVENT.
Configdialog shows a `notifiy` if there is a `notify_func` with SOURCE_BOTTOM_WHATSOEVER end the emits the events with SOURCE_EVENT.

I will put all together, so we can talk about it.

Reply:
(If you're tweaking bitmasks, update your 1 2 4 16 32... to read as 0x0001, 0x0010... - makes it easier to follow:
https://github.com/koreader/koreader/blob/247166b3df4e226f35b2a225e5ae21d14e7b42a0/frontend/apps/reader/modules/readertypeset.lua#L347-L352
)